File: frontend/src/lib/api/availability.ts
import { client } from "./client";

export interface Poll {
  id: number;
  match_id: number;
  deadline: string | null;
  created_at: string;
}

export interface AvailabilityResponse {
  id: number;
  poll_id: number;
  user_id: number;
  status: string;
  responded_at: string;
}

export interface PollWithResponses {
  poll: Poll;
  responses: AvailabilityResponse[];
  available: number;
  unavailable: number;
  maybe: number;
}

export const availabilityApi = {
  createPoll: (matchId: number, deadline?: string) =>
    client.post<Poll>("/availability/polls", { match_id: matchId, deadline }).then((r) => r.data),

  getPoll: (matchId: number) =>
    client.get<PollWithResponses>(`/availability/polls/${matchId}`).then((r) => r.data),

  respond: (matchId: number, status: string) =>
    client.post<AvailabilityResponse>(`/availability/polls/${matchId}/respond`, { status }).then((r) => r.data),
};

File: backend/app/schemas/availability.py
from __future__ import annotations

from datetime import datetime

from pydantic import BaseModel, ConfigDict


class CreatePollRequest(BaseModel):
    match_id: int
    deadline: datetime | None = None


class SubmitAvailabilityRequest(BaseModel):
    status: str  # available / unavailable / maybe


class PollResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    match_id: int
    deadline: datetime | None
    created_at: datetime


class AvailabilityResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    poll_id: int
    user_id: int
    status: str
    responded_at: datetime


class PollWithResponses(BaseModel):
    poll: PollResponse
    responses: list[AvailabilityResponse]
    available: int
    unavailable: int
    maybe: int
